Examples/Algorithms/Geant4/src/SensitiveSurfaceMapper.cpp

```diff
@@ -71,11 +71,16 @@ void ActsExamples::SensitiveSurfaceMapper::remapSensitiveNames(
 
   std::string volumeName = g4LogicalVolume->GetName();
   std::string volumeMaterialName = g4LogicalVolume->GetMaterial()->GetName();
-  if (g4SensitiveDetector == nullptr ||
+
+  bool isSensitive = g4SensitiveDetector != nullptr;
+  bool isMappedMaterial =
       std::find(m_cfg.materialMappings.begin(), m_cfg.materialMappings.end(),
-                volumeMaterialName) != m_cfg.materialMappings.end() ||
+                volumeMaterialName) != m_cfg.materialMappings.end();
+  bool isMappedVolume =
       std::find(m_cfg.volumeMappings.begin(), m_cfg.volumeMappings.end(),
-                volumeName) != m_cfg.volumeMappings.end()) {
+                volumeName) != m_cfg.volumeMappings.end();
+
+  if (isSensitive && (isMappedMaterial || isMappedVolume)) {
```

Comment:
I might be completely wrong here, but wouldn't this mean that surfaces are never mapped if g4SensitiveDetector == nullptr?
Wouldn't you want this to run in this case?
Considering the message at the end maybe it has to be (isSensitive || isMappedMaterial || isMappedVolume) after all.
I think I also got confused along the way

Reply:
right :+1: